iosApp/iosApp/Pages/Map/HomeMapViewHandlerExtension.swift

```diff
@@ -215,37 +125,28 @@ extension HomeMapView {
 
     func handleTapVehicle(_ vehicle: Vehicle) {
         guard let tripId = vehicle.tripId else { return }
-        guard case .stopDetails = nearbyVM.navigationStack.lastSafe() else { return }
-
-        let routeCardData = nearbyVM.routeCardData
-        let route = routeCardData?.first(where: { $0.lineOrRoute.containsRoute(routeId: vehicle.routeId) })
-        let stop = route?.stopData.first
-        let allTrips = stop?.data.flatMap(\.upcomingTrips)
-        let trip = allTrips?.first(where: { upcoming in
-            upcoming.trip.id == tripId
-        })
-        let stopSequence = trip?.stopSequence
-        let routeId = trip?.trip.routeId ?? vehicle.routeId ?? route?.lineOrRoute.id
+        guard case let .stopDetails(stopId, stopFilter, tripFilter) = nearbyVM.navigationStack.lastSafe()
+        else { return }
+        guard stopFilter != nil || tripFilter?.tripId == tripId else { return }
+        let routeCard = nearbyVM.routeCardData?.first(where: { $0.lineOrRoute.containsRoute(routeId: vehicle.routeId) })
+        let upcoming = routeCard?
+            .stopData
+            .flatMap(\.data)
+            .flatMap(\.upcomingTrips)
+            .first { upcoming in upcoming.trip.id == tripId }
+        let stopSequence = upcoming?.stopSequence
+        let routeId = upcoming?.trip.routeId ?? vehicle.routeId ?? routeCard?.lineOrRoute.id
 
         if let routeId { analytics.tappedVehicle(routeId: routeId) }
 
-        nearbyVM.navigationStack.lastTripDetailsFilter = .init(
+        let newTripFilter = TripDetailsFilter(
             tripId: tripId,
             vehicleId: vehicle.id,
             stopSequence: stopSequence,
             selectionLock: true
         )
-    }
-
-    func handleSelectedVehicleChange(_ previousVehicle: Vehicle?, _ nextVehicle: Vehicle?) {
-        guard let globalData = mapVM.globalData else { return }
-        guard let stop = nearbyVM.getTargetStop(global: globalData) else { return }
-        guard let nextVehicle else {
-            viewportProvider.animateTo(coordinates: stop.coordinate)
-            return
-        }
-        if previousVehicle == nil || previousVehicle?.id != nextVehicle.id {
-            viewportProvider.vehicleOverview(vehicle: nextVehicle, stop: stop)
-        }
+        let stop = globalData?.getStop(stopId: stopId)
+        nearbyVM.navigationStack.lastTripDetailsFilter = newTripFilter
+        mapVM.selectedTrip(stopFilter: stopFilter, stop: stop, tripFilter: newTripFilter, vehicle: vehicle)
```

Comment:
question: If the navChanged, is it necessary to also call `mapVM.selectedTrip`? Is`MapVM.navChanged` already called somewhere?